Source/FullScreenPokemon.ts

```diff
@@ -9334,25 +9358,49 @@ module FullScreenPokemon {
         }
 
         /**
-         * Saves the position of a certain Character.
-         * 
-         * @param FSP
-         * @param character   An in-game Character.
-         * @param id   The ID associated with the Character.
+         * Clears the saved data in localStorage upon a new game being started and saves it
+         * in localStorage.
          */
-        saveCharacterPosition(FSP: FullScreenPokemon, character: ICharacter, id: string): void {
-            FSP.StateHolder.addChange(
-                id,
-                "xloc",
-                (character.left + FSP.MapScreener.left) / FSP.unitsize);
-            FSP.StateHolder.addChange(
-                id,
-                "yloc",
-                (character.top + FSP.MapScreener.top) / FSP.unitsize);
-            FSP.StateHolder.addChange(
-                id,
-                "direction",
-                character.direction);
+        clearSavedData(): void {
+            let oldLocalStorage: any = this.ItemsHolder.exportItems();
+
+            let collectionKeys: string[] = this.ItemsHolder.getItem(this.StateHolder.getPrefix() + "collectionKeys");
+            for (let i: number = 0; collectionKeys && i < collectionKeys.length; i += 1) {
+                oldLocalStorage[collectionKeys[i]] = this.ItemsHolder.getItem(collectionKeys[i]);
+            }
+
+            let keys: string[] = this.ItemsHolder.getKeys();
+            for (let i: number = 0; i < keys.length; i += 1) {
+                delete this.ItemsHolder.getLocalStorage()[this.ItemsHolder.getPrefix() + keys[i]];
```

Comment:
Actually a clarification, you mean in regards to not calling it in the loop, right?